### PR TITLE
Map by country

### PR DESCRIPTION
**Pull request by squirrelo**
This uses D3 to create the map at a country level. Please test this out and make sure it responds on other computers than mine, as well as color scheme and any other suggestions.

### CODE REVIEW

**Comment by squirrelo** [comment]
Also, the text is all placeholder until this is finalized, then it will be added to the appropriate dictionaries.

**Comment by wasade** [comment]
does this need the test db?

File: amgut/handlers/change_pass_verify.py [hunk 1]
```diff
@@ -53,9 +53,9 @@ def reset_pass_and_email(self, new_password, confirm_password, email,
             self.render('change_pass_verify.html', email='', kitid='',
                         passocde='', new_password='',
                         confirm_password='', result=4, message='',
```
Comment (wasade) [inline amgut/handlers/change_pass_verify.py]: `result=4`?

File: amgut/lib/data_access/ag_data_access.py [hunk 2]
```diff
@@ -435,50 +435,9 @@ def verifyKit(self, supplied_kit_id):
...
-        # set blue for unverified kits
-        res.update({ll[kid]: '00B2FF' for kid in notverified if kid in ll})
-
-        # set yellow for all others
-        res.update({v: 'FFFF00' for k, v in ll.items() if v not in res})
-
-        return [[lat, lng, c] for ((lat, lng), c) in res.items()]
+        sql = """SELECT country, count(country)::integer
+                 FROM ag.ag_login GROUP BY country"""
+        return dict(self._sql.execute_fetchall(sql))
```
Comment (wasade) [inline amgut/lib/data_access/ag_data_access.py]: so much cleaner

File: amgut/templates/map.html [hunk 3]
```diff
@@ -1,28 +1,146 @@
...
-</div>
+d3.select(window).on("resize", throttle);
+
+var zoom = d3.behavior.zoom()
+    .scaleExtent([1, 9])
+    .on("zoom", move);
+
+var width = document.getElementById('map_canvas').offsetWidth;
+var height = document.getElementById('map_canvas').offsetHeight;
+var topo,projection,path,svg,g;
```
Comment (wasade) [inline amgut/templates/map.html]: spaces please

File: amgut/templates/sitebase.html [hunk 4]
```diff
@@ -32,141 +32,137 @@
...
-            {{sitebase_barcode['barcode']}}
-            {% if sitebase_barcode['status'] == 'Received' %}
-                {% raw media_locale['NAV_RECEIVED'] %}
-            {% end %}
-                                        </span>
-                                    </a>
-                                </li>
+                    <li class="last">
+                        <a onclick="document.forms['participant_overview_{{ap}}'].submit(); return false;">
+                            <form action="{% raw media_locale['SITEBASE'] %}/participants/{{ap}}" method="post" id="participant_overview_{{ap}}">
```
Comment (wasade) [inline amgut/templates/sitebase.html]: `/participants/` should be under authed, right?

Comment (squirrelo) [inline amgut/templates/sitebase.html]: this is the authed sitebase, so should be OK.

**Comment by squirrelo** [comment]
changes made

File: amgut/templates/sitebase.html (hunk 4 above)
Comment (wasade) [inline amgut/templates/sitebase.html]: right, but it's inconsistent? If you agree, i'll make an issue

**Comment by wasade** [comment]
I dont understand the comment about "all test is placeholder", is this PR good for review (and merge if okay)?

File: amgut/templates/sitebase.html (hunk 4 above)
Comment (squirrelo) [inline amgut/templates/sitebase.html]: Oh, I see what you mean. Yeah, can open an issue.

**Comment by squirrelo** [comment]
The text there is placeholder right now, not pulled out of the locale dictionaries. Once we agree the colors & text are good I was going to dump the text in them.

**Comment by squirrelo** [comment]
Ready for review

**Comment by squirrelo** [comment]
ping @biocore/american-gut-web-admin

**Comment by squirrelo** [comment]
ping @EmbrietteH @wasade @mortonjt

File: amgut/handlers/change_pass_verify.py (hunk 1 above)
Comment (mortonjt) [inline amgut/handlers/change_pass_verify.py]: ping on this @squirrelo . Where does this magic number come from?

Comment (squirrelo) [inline amgut/handlers/change_pass_verify.py]: Old code that shows different messages depending on number. Outside of this PR scope to change.

Comment (mortonjt) [inline amgut/handlers/change_pass_verify.py]: You don't have to change it - but you should probably add a comment on where this came from.

Comment (squirrelo) [inline amgut/handlers/change_pass_verify.py]: Please open an issue and it can be changed as part of the issue.

Comment (mortonjt) [inline amgut/handlers/change_pass_verify.py]: [Done](#551)

Comment (squirrelo) [inline amgut/handlers/change_pass_verify.py]: Thanks!

**Comment by mortonjt** [comment]
I'm seeing some really weird rendering problems.  Maybe we should step through this PR once you get back?

**Comment by squirrelo** [comment]
Should all be fixed.

**Comment by mortonjt** [comment]
Looks fixed to me.

**Comment by squirrelo** [comment]
cool, good to merge then?

**Pull request merged**